config/environments/production.rb

```diff
@@ -109,4 +109,17 @@
   # config.active_record.database_selector = { delay: 2.seconds }
   # config.active_record.database_resolver = ActiveRecord::Middleware::DatabaseSelector::Resolver
   # config.active_record.database_resolver_context = ActiveRecord::Middleware::DatabaseSelector::Resolver::Session
+  config.action_mailer.delivery_method = :smtp
+  host = 'https://rails-stock-trading-app.herokuapp.com/' #replace with your own url
+  config.action_mailer.default_url_options = { host: host }
+
+  # SMTP settings for gmail
+  config.action_mailer.smtp_settings = {
+    :address              => "smtp.gmail.com",
+    :port                 => 587,
+    :user_name            => <gmail_username>,
+    :password             => <gmail_password>,
```

Comment:
how did you store the value of `<gmail_username>` and  `<gmail_password>`?

Reply:
hi maam @analyn-cajocson , this was stored in credentials with the thought process that it should be encrypted or hidden. Was this the right thing to do or is it advisable to have it written there in the production.rb itself?

Reply:
Hmm maybe you are not doing it right. Have you checked to `rails console` if you can access the `<gmail_username> `and `<gmail_password>`?

Reply:
I think I am not doing it right. I followed [this site](https://dev.to/morinoko/sending-emails-in-rails-with-action-mailer-and-gmail-35g4) and it said "Replace <gmail_username> and <gmail_password> with your own username and password, which would preferably be hidden away as environment variables." I might have assumed to edit it in credentials. I'll have it written in the production.rb instead

Reply:
Refer to this: https://blog.saeloun.com/2019/10/10/rails-6-adds-support-for-multi-environment-credentials.html#how-to-read-credentials

Reply:
Hi maam @analyn-cajocson , based on the first commit I am able to access the rails console with my gmail username and pw. Will this do?

![image](https://user-images.githubusercontent.com/81552806/134444608-c6d89d63-6fc5-45a1-aed9-45478bf48dd4.png)

Reply:
You should do that to access your credentials inside your app hehe